app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,45 @@
+module SessionsHelper
+  # Logs in the given user.
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  # Returns the current logged-in user (if any).
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
+  end
```

Comment:
I feel this can be refactored into something neater.

Something like this (untested):

```ruby
def current_user
  current_user_from_session || current_user_from_cookies
end

private

def current_user_from_session
  @current_user = User.find_by(id: session[:user_id])
end

def current_user_from_cookies
  user = User.find_by(id: cookies.signed[:user_id])

  if user && user.authenticated?(cookies[:remember_token])
    log_in user
    @current_user = user
  end
end
```

The benefits of this over the current code:

+ Smaller methods
+ Method names tell a dev what they are doing
+ No nested `ifs` (code smell)